src/app/games/diaballik/DiaballikMoveGenerator.ts

```diff
@@ -122,27 +137,29 @@ export class DiaballikMoveGenerator extends MoveGenerator<DiaballikMove, Diaball
                 } else {
                     // Can only do two translations per move
                     const hasLessThanTwoTranslations: boolean = moveInConstruction.translations < 2;
-                    // If there was a pass, we ignore other pieces than the one that has made the pass
-                    // The reasoning is that any other translation could have been done before the pass,
-                    // hence they are not interesting to generate.
-                    const ifHasPassThenIsPieceThatPassed: boolean =
-                        moveInConstruction.hasPass === false || moveInConstruction.getPassEnd().equalsValue(coord);
-                    if (hasLessThanTwoTranslations && ifHasPassThenIsPieceThatPassed) {
-                        let forbiddenEnd: MGPOptional<Coord> = MGPOptional.empty();
-                        if (moveInConstruction.translations > 0) {
-                            const previousTranslation: DiaballikTranslation =
-                                moveInConstruction.getPreviousTranslation().get();
-                            if (previousTranslation.getEnd().equals(coord)) {
-                                forbiddenEnd = MGPOptional.of(previousTranslation.getStart());
-                            }
-                        }
+                    if (hasLessThanTwoTranslations) {
+                        const forbiddenEnd: MGPOptional<Coord> =
+                            this.getPreviousTranslationStartThatEndsAt(moveInConstruction, coord);
                         for (const end of this.getTranslationEnds(state, coordAndContent.coord)) {
                             // We ignore the forbidden end because it is the opposite
                             // of the previous translation, hence it a no-op
-                            if (forbiddenEnd.equalsValue(end) === false) {
-                                // The translation here is valid and legal by construction
-                                const translation: DiaballikTranslation = DiaballikTranslation.from(coord, end).get();
-                                moveInConstruction.addIfLegal(translation, nextMovesInConstruction);
+                            if (forbiddenEnd.equalsValue(end) === false || this.avoidDuplicates === false) {
```

Comment:
100% de ce qui se passe dans le for dépend de la faussetté de avoidDuplicates, on peut le mettre en dehors de la loup du coup ?

Reply:
non ! Le avoidDulicates est là pour activer le mode "osef des duplicatas, donne moi tous les moves comme un idiot". Mais "le sortir de la boucle" changerait le comportement. C'était plus pour tester ce que ça change de filtrer, donc ptet que maintenant ça peut partir, vu que c'est mieux de les filtrer et qu'on voudra pas activer avoidDuplicates ?

Reply:
Yep à terme avoidDuplicate est plutôt du genre à devoir être vrai tout le temps, on fait le skip ?

En revanche ouais ma lecture de if était foireuse haha :D

Reply:
il est vrai tout le temps SAUF pour afficher les indicateurs, donc on veut pas l'enlever je crois !

Reply:
On a ticketté l'idée d'avoir GameNode qui refuse de dupliquer ses enfants qui ont le même state, quitte à considérer que un ou plusieurs mouvement sont les "node.previousMove" ?

Reply:
répondu dans trello ! Tu ne peux pas résoudre :D !

Reply:
ticket créé !